src/pipeline_migration/utils.py

```diff
@@ -1,17 +1,100 @@
+from dataclasses import dataclass, field
 from typing import Any
-from ruamel.yaml import YAML
 
 from pipeline_migration.types import FilePath
 
+from ruamel.yaml import YAML, CommentedMap, CommentedSeq
+from ruamel.yaml.comments import CommentedBase
+
+
+__all__ = [
+    "BlockSequenceIndentation",
+    "create_yaml_obj",
+    "dump_yaml",
+    "is_true",
+    "load_yaml",
+    "YAMLStyle",
+]
+
 
 def is_true(value: str) -> bool:
     return value.strip().lower() == "true"
 
 
-def create_yaml_obj():
+@dataclass
+class BlockSequenceIndentation:
+    indentations: dict[int, int] = field(default_factory=dict)
+
+    @property
+    def is_consistent(self) -> bool:
+        return len(self.levels) == 1
+
+    @property
+    def levels(self) -> list[int]:
+        return list(self.indentations.keys())
+
+
+@dataclass
+class _NodePath:
+    node: CommentedBase
+    field: str = ""
```

Comment:
nitpick: I think the `field` attribute is unused, which kinda makes this class unnecessary

Reply:
The `node` is required for determine the indentation levels.